sdk/eventhub/event-hubs/tests.yml

```diff
@@ -6,22 +6,57 @@ jobs:
       PackageName: "@azure/event-hubs"
       # Remove Browser tests from matrix since they are currently a no-op
       Matrix:
+        Linux_Node8X:
+          OSVmImage: "ubuntu-16.04"
+          TestType: "node"
+          NodeVersion: "8.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-lin8)
+        Windows_Node8X:
+          OSVmImage: "vs2017-win2016"
+          TestType: "node"
+          NodeVersion: "8.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-win8)
+        Mac_Node8X:
+          OSVmImage: "macOS-10.13"
+          TestType: "node"
+          NodeVersion: "8.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-mac8)
         Linux_Node10X:
           OSVmImage: "ubuntu-16.04"
           TestType: "node"
+          NodeVersion: "10.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-lin10)
         Windows_Node10X:
           OSVmImage: "vs2017-win2016"
           TestType: "node"
+          NodeVersion: "10.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-win10)
         Mac_Node10X:
           OSVmImage: "macOS-10.13"
           TestType: "node"
+          NodeVersion: "10.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-mac10)
+        Linux_Node12X:
+          OSVmImage: "ubuntu-16.04"
+          TestType: "node"
+          NodeVersion: "12.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-lin12)
+        Windows_Node12X:
+          OSVmImage: "vs2017-win2016"
+          TestType: "node"
+          NodeVersion: "12.x"
+          EVENTHUB_NAME: $(js-event-hubs-test-name-win12)
```

Comment:
I don't believe this will set this in the environment.

Reply:
Actually, Mike had referenced this link for this : https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=example#matrix
In this example, isn't PYTHON_VERSION an environment var?

Reply:
https://docs.microsoft.com/en-us/azure/devops/pipelines/migrate/from-travis?view=azure-devops#example-environment-variable-in-a-matrix

Reply:
I think it so but there is definitely some odd behaviors with the way DevOps handles variables being promoted to environment variables, especially in the case of secrets. I'd just suggest you double check to ensure the variables are set as you'd expect.

Reply:
The templates don't presently support environment variables in the matrix. In the linked documentation `PYTHON_VERSION` is just a pipeline variable and can be referenced with `$(PYTHON_VERSION)`... 

The current template was designed around a global definition of environment variables instead of matrix-entry-specific environment variables. 

It might be possible in `archetype-sdk-tests.yml` to combine environment variables from the template parameters and entries in the matrix.

Reply:
@danieljurek Going through the `archetype-sdk-tests.yml` I understand that the Env Vars are being pulled in from `tests-common.yml` . So now we need a way of telling it to pull all the EnvVars from tests-common.yml except one `EVENTHUB_NAME` be pulled from the matrix-entry

Reply:
As discussed today in the standup, I updated the `EVENTHUB_NAME` so that the value is specified within the matrix instead being pulled as a secret. And that does automatically configure it as an env variable. When I test this, all the tests can run in parallel and they all pass. :-)

So I think the issue was that a secret was not being set as an env variable when specified for each matrix entry. Not sure if this is a devops bug that we should file or is this an expected behaviour 
cc: @mitchdenny @weshaggard @mikeharder @danieljurek

Reply:
Yes if you hard code the values in the yml I do believe those will correctly be set as env variables.

Note if you already added these into keyvault please clean up those values if we aren't planning to use them.

Reply:
Yes cleaned up

Reply:
From my comment above: I was mistaking secret variable handling with job/matrix variables. Sure enough in the documentation is this sentence: 
> Notice that variables are also made available to scripts through environment variables.

(https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch)

Thanks for patiently walking me through how you got the right outputs! :) 

One possible alternative (not tested yet) to using secrets in those places is to use a variable group that is not pulling secrets from KV. The variables might populate properly but I haven't tested that behavior to date.

Reply:
There is no need to create a variable group unless we need to share or hook them up to keyvault. For this case I think having them be hard coded in the yml file is probably the best way to configure them.

Reply:
Thanks @danieljurek and @weshaggard